File: back/back/Models/Grade.cs
using LiteDB;

namespace back.Models;

public sealed class Grade
{
    [BsonId]
    public int Id { get; set; }

    public string Nom { get; set; } = null!;
    public string? NomFichierIcone { get; set; }
    public string? Fonction { get; set; }

    public int Ordre { get; set; }
    public int NbPlace { get; set; }
    public int NbOperationRequis { get; set; }
     public int NbPointBoutiqueGagnerParOperation { get; set; }

     public int Conserne { get; set; }
    public bool CandidatureRequise { get; set; }
    public bool EstHonorifique { get; set; }
}

File: front/src/models/Grade.ts
export type Grade =
{
    id: number,
    nom: string,
    nomRaccourci: string,
    urlFichierIcone: string | null,
    fonction: string | null,
    ordre: number,
    nbPlace: number,
    nbOperationRequis: number,
    nbPointBoutiqueGagnerParOperation: number,
    conserne: number,
    candidatureRequise: boolean,
    estHonorifique: boolean
}

export type GradeLeger =
{
    id: number,
    nom: string,
    nomRaccourci: string
}

export type GradeRequete = Omit<Grade, "id"|"nomFichier">;